junit-platform-engine/src/main/java/org/junit/platform/engine/support/store/Namespace.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright 2015-2025 the original author or authors.
+ *
+ * All rights reserved. This program and the accompanying materials are
+ * made available under the terms of the Eclipse Public License v2.0 which
+ * accompanies this distribution and is available at
+ *
+ * https://www.eclipse.org/legal/epl-v20.html
+ */
+
+package org.junit.platform.engine.support.store;
+
+import static org.apiguardian.api.API.Status.EXPERIMENTAL;
+import static org.apiguardian.api.API.Status.STABLE;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+import org.apiguardian.api.API;
+import org.junit.platform.commons.util.Preconditions;
+
+@API(status = EXPERIMENTAL, since = "5.13")
+public class Namespace {
+
+	/**
+	 * The default, global namespace which allows access to stored data from
+	 * all extensions.
+	 */
+	public static final Namespace GLOBAL = Namespace.create(new Object());
```

Comment:
I think `ExtensionContext.Namespace.GLOBAL` needs to be created from this in order for the conversion in `LauncherStoreFacade` to work correctly and we should add a test for that.

Reply:
see https://github.com/junit-team/junit5/pull/4281#pullrequestreview-2671378936